### ISSUE TITLE
RH SSO offline token causing Violation of PRIMARY KEY constraint 'CONSTRAINT_OFFL_CL_SES_PK3'

### ISSUE THREAD

**Issue by mposolda** [keycloak/keycloak#13706]
### Describe the bug

Customer is seeing the following in the logs after upgrading to 7.5.2.
When their apps are refreshing the access token using the offline tokens, they are noticing 'CONSTRAINT_OFFL_CL_SES_PK3' violation on the OFFLINE_CLIENT_SESSION table.

Logs:

[org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (default task-173) Violation of PRIMARY KEY constraint 'CONSTRAINT_OFFL_CL_SES_PK3'. Cannot insert duplicate key in object 'dbo.OFFLINE_CLIENT_SESSION'. The duplicate key value is (<some-uuid>, <another-uuid>, local, local, 1).

Customer upgraded RHSSO from 7.3.8 to 7.5.0 on June 2nd and then patched it to 7.5.2.

Customer is looking to find out why this error happens?

Customer investigated and believes this is
```
    happening when the client session is expired in infinispan cache but is not yet removed in the
offline_client_session table. That is when I believe it will try to make an entry into offline_client_session
table with the existing user_sessoion_id and client_id
```

### Version

19.0.1

### Anything else?

This comes from the support ticket 281

**Comment by mposolda** [comment]
Adding to milestone 20 for now, but it can be possibly postponed further if we don't have capacity.

**Comment by marcelomrwin** [comment]
We performed tests with version 7.5.2 simulating the mentioned situation and observed that the session was removed from the database as expected. Refresh tokens were used normally.

We also performed the same tests using version 19.0.2 of keycloak as a base. Even in cases where the session had not yet been deleted from the database and the refresh token was used normally.

An unobserved scenario might be version upgrades as mentioned in the issue. In this scenario, if something happens, it could point to a failure during the patching process. However, using "clean" versions we do not observe this behavior.
